src/nvim/api/buffer.c

```diff
@@ -737,6 +741,216 @@ ArrayOf(Integer, 2) nvim_buf_get_mark(Buffer buffer, String name, Error *err)
   return rv;
 }
 
+/// Returns mark info at the given position or mark index
+///
+/// @param buffer The buffer handle
+/// @param namespace a identifier returned previously with extmark_ns_create
+/// @param id (row, col) or mark_id of mark
+/// @param[out] err Details of an error that may have occurred
+/// @return [mark_id, row, col] list
+ArrayOf(Object) nvim_buf_lookup_mark(Buffer buffer,
+                                     Integer namespace,
+                                     Object index,
```

Comment:
For this function it only makes sense for index to be `Integer`. Otherwise `get_marks` should be used.